rand_distr/src/normal.rs

```diff
@@ -140,23 +144,57 @@ impl std::error::Error for Error {}
 impl<F> Normal<F>
 where F: Float, StandardNormal: Distribution<F>
 {
-    /// Construct a new `Normal` distribution with the given mean and
-    /// standard deviation.
+    /// Construct, from mean and standard deviation
+    ///
+    /// Parameters:
+    ///
+    /// -   mean (`μ`, unrestricted)
+    /// -   standard deviation (`σ`, must be finite)
     #[inline]
     pub fn new(mean: F, std_dev: F) -> Result<Normal<F>, Error> {
-        if !(std_dev >= F::zero()) {
-            return Err(Error::StdDevTooSmall);
+        if !std_dev.is_finite() {
+            return Err(Error::BadVariance);
         }
         Ok(Normal { mean, std_dev })
     }
+
+    /// Construct, from mean and coefficient of variation
+    ///
+    /// Parameters:
+    ///
+    /// -   mean (`μ`, unrestricted)
+    /// -   coefficient of variation (`cv = abs(σ / μ)`)
+    #[inline]
+    pub fn from_mean_cv(mean: F, cv: F) -> Result<Normal<F>, Error> {
+        if !cv.is_finite() || cv < F::zero() {
+            return Err(Error::BadVariance);
+        }
+        let std_dev = cv * mean;
```

Comment:
This gives nonsensical results for `mean == 0`. Maybe we should return an error for that?

Reply:
As I understand, for any finite CV and mean=0 samples should simply be 0?

Reply:
But CV is not finite for `mean == zero`. There are a lot of normal distributions with zero mean and non-zero variance.

Reply:
If CV is not finite then the above condition will already catch this case. If the user attempts to parametrise a normal with mean and CV when the mean may be zero, that's already their problem I think? Using CV is only appropriate when the variance *should be* proportional to the size of the value — and may often be more appropriate for log-normal than normal.

Reply:
> If the user attempts to parametrise a normal with mean and CV when the mean may be zero, that's already their problem I think?

What's wrong with returning an error in this case?

Reply:
It would be perfectly valid to use `Normal::from_mean_cv(x, CV)` where `x > 0` and `CV` is fixed. Why is it not valid to extend this to `x = 0`? The result is statistically unambiguous.

If we allowed non-finite CV this would not be the case, but non-finite CV is *never* valid.

Reply:
Ok, fair enough. It makes sense as a limit.